### PR TITLE
Handle404

### PR DESCRIPTION
**Pull request by livike**

### CODE REVIEW

File: web/templates/pages/project_edit.html [hunk 1]
```diff
@@ -6,11 +6,17 @@
 {% endblock %}
 
 {% block content %}
-	<h1 class="pull-left">{% if editing %}Edit{% else %}Add {% endif %} your project</h1>
-	<form method="POST">{% csrf_token %}
-		{{ projectform.as_p }}
-		<button type="submit" class="save btn btn-default">Save</button>
-	</form>
+
+	{% if project and editing %}
```
Comment (jurecuhalev) [inline web/templates/pages/project_edit.html]: This rule looks wrong. "If project is defined, display an error"?

File: web/templates/pages/project_detail.html [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
+				<p><strong>Repository:</strong> <a href="{{ project.repository_link }}"> {{ project.repository_link }}</a></p>
+				{% if project.website_production %}
+					<p><strong>Production site:</strong> <a href="{{ project.website_production }}"> {{ project.website_production }}</a></p>
+				{% endif %}
+				{% if project.website_test %}
+					<p><strong>Test site:</strong> <a href="{{ project.website_test }}"> {{ project.website_test }}</a></p>
+				{% endif %}
+			</div>
+		</div>	
+		<div class="row">
```
Comment (jurecuhalev) [inline web/templates/pages/project_detail.html:32]: You're missing `<div class="col-md-8 col-xs-6">` column classes after .row

File: web/templates/pages/project_edit.html (hunk 1 above)
Comment (livike) [inline web/templates/pages/project_edit.html]: Seems wrong but try to add  {% if not project and editing %} display the warning also

Comment (gandalfar) [inline web/templates/pages/project_edit.html]: You need it twice: if not project and not editing

File: web/views.py [hunk 3]
```diff
@@ -54,7 +58,10 @@ def project_new(request):
 	return render(request, 'pages/project_edit.html', {'projectform': projectform})
 
 def project_edit(request, project_id):
-	project = get_object_or_404(Project, project_id=project_id)
+	if Project.objects.filter(pk=project_id).exists():
+		project = Project.objects.get(pk=project_id)
+	else:
+		project = None
```
Comment (jurecuhalev) [inline web/views.py]: Instead of Project = None, just `return`:
`return render(request, 'pages/project_edit.html', {'invalid_project_id': True})`

**Pull request closed without merging**